src/test/java/org/sagebionetworks/bridge/spring/controllers/MasterSchedulerControllerTest.java

```diff
@@ -0,0 +1,242 @@
+import static org.sagebionetworks.bridge.TestConstants.TEST_STUDY_IDENTIFIER;
+import static org.sagebionetworks.bridge.TestUtils.assertCreate;
+import static org.sagebionetworks.bridge.TestUtils.assertCrossOrigin;
+import static org.sagebionetworks.bridge.TestUtils.assertDelete;
+import static org.sagebionetworks.bridge.TestUtils.assertGet;
+import static org.sagebionetworks.bridge.TestUtils.assertPost;
+import static org.sagebionetworks.bridge.TestUtils.mockRequestBody;
+
+import org.joda.time.DateTime;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+
+import org.sagebionetworks.bridge.TestUtils;
+import org.sagebionetworks.bridge.dynamodb.DynamoStudy;
+import org.sagebionetworks.bridge.exceptions.UnauthorizedException;
+import org.sagebionetworks.bridge.models.ResourceList;
+import org.sagebionetworks.bridge.models.StatusMessage;
+import org.sagebionetworks.bridge.models.TimestampHolder;
+import org.sagebionetworks.bridge.models.accounts.StudyParticipant;
+import org.sagebionetworks.bridge.models.accounts.UserSession;
+import org.sagebionetworks.bridge.models.schedules.MasterSchedulerConfig;
+import org.sagebionetworks.bridge.models.studies.Study;
+import org.sagebionetworks.bridge.services.MasterSchedulerService;
+import org.sagebionetworks.bridge.services.StudyService;
+
+public class MasterSchedulerControllerTest extends Mockito {
+    private static final long TIME_STAMP = DateTime.parse("2018-03-27T18:30-07:00").getMillis();
+    private static final String SCHEDULE_ID = "test-schedule-id";
+    private static final String CRON_SCHEDULE = "testCronSchedule";
+    private static final String UPDATED_CRON_SCHEDULE = "updatedCronSchedule";
+    private static final String SQS_QUEUE_URL = "testSysQueueUrl";
+    private static final String UPDATED_SQS_QUEUE_URL = "updatedSysQueueUrl";
+    private static final long VERSION = 1L;
+    
+    @Mock
+    MasterSchedulerService mockSchedulerService;
+    
+    @Mock
+    StudyService mockStudyService;
+    
+    @Mock
+    HttpServletRequest mockRequest;
+    
+    @Mock
+    HttpServletResponse mockResponse;
+    
+    @Captor
+    ArgumentCaptor<MasterSchedulerConfig> configCaptor;
+    
+    @Spy
+    @InjectMocks
+    MasterSchedulerController controller = new MasterSchedulerController();
+    
+    MasterSchedulerConfig mockConfig;
+    
+    UserSession session; 
+    
+    Study study;
+    
+    @BeforeMethod
+    private void before() {
+        MockitoAnnotations.initMocks(this);
+        
+        mockConfig = TestUtils.getMasterSchedulerConfig();
+        
+        study = new DynamoStudy();
```

Comment:
We should use Study.create(), so the Dynamo details don't leak outside the Dynamo classes.